core/arch/aarch64/instr.c

```diff
@@ -335,6 +335,12 @@ reg_is_gpr(reg_id_t reg)
     return (DR_REG_X0 <= reg && reg <= DR_REG_WSP);
 }
 
+bool
+reg_is_simd(reg_id_t reg)
+{
+    return (DR_REG_Q0 <= reg && reg <= DR_REG_B31);
```

Comment:
I am a bit surprised that `S`, `H` and `B` registers are included here. I thought the only vector registers used by instructions are the 128 bit `Q`, as in `v8.2d, v8.4s, v8.8h` and 64 bit `D` as in `v8.2s`, `v8.4h`. Am I missing something and are there encodings that use `S`, `H` or `B` registers as vector register?

Reply:
My (simple) understanding is that all `reg_is_simd()` is saying is whether a given register is SIMD or not, and `B`/`H`/`S`/`D` are components of a SIMD `Q` register. That's independent of any encoding issues. Have I missed something?